airflow/kubernetes/pod_launcher.py

```diff
@@ -14,283 +15,15 @@
-                namespace=pod.metadata.namespace, field_selector=f"involvedObject.name={pod.metadata.name}"
-            )
-        except BaseHTTPError as e:
-            raise AirflowException(f'There was an error reading the kubernetes API: {e}')
-
-    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
-    def read_pod(self, pod: V1Pod):
-        """Read POD information"""
-        try:
-            return self._client.read_namespaced_pod(pod.metadata.name, pod.metadata.namespace)
-        except BaseHTTPError as e:
-            raise AirflowException(f'There was an error reading the kubernetes API: {e}')
-
-    def _extract_xcom(self, pod: V1Pod):
-        resp = kubernetes_stream(
-            self._client.connect_get_namespaced_pod_exec,
-            pod.metadata.name,
-            pod.metadata.namespace,
-            container=PodDefaults.SIDECAR_CONTAINER_NAME,
-            command=['/bin/sh'],
-            stdin=True,
-            stdout=True,
-            stderr=True,
-            tty=False,
-            _preload_content=False,
-        )
-        try:
-            result = self._exec_pod_command(resp, f'cat {PodDefaults.XCOM_MOUNT_PATH}/return.json')
-            self._exec_pod_command(resp, 'kill -s SIGINT 1')
-        finally:
-            resp.close()
-        if result is None:
-            raise AirflowException(f'Failed to extract xcom from pod: {pod.metadata.name}')
-        return result
-
-    def _exec_pod_command(self, resp, command):
-        if resp.is_open():
-            self.log.info('Running command... %s\n', command)
-            resp.write_stdin(command + '\n')
-            while resp.is_open():
-                resp.update(timeout=1)
-                if resp.peek_stdout():
-                    return resp.read_stdout()
-                if resp.peek_stderr():
-                    self.log.info(resp.read_stderr())
-                    break
-        return None
-
-    def process_status(self, job_id, status):
-        """Process status information for the JOB"""
-        status = status.lower()
-        if status == PodStatus.PENDING:
-            return State.QUEUED
-        elif status == PodStatus.FAILED:
-            self.log.error('Event with job id %s Failed', job_id)
-            return State.FAILED
-        elif status == PodStatus.SUCCEEDED:
-            self.log.info('Event with job id %s Succeeded', job_id)
-            return State.SUCCESS
-        elif status == PodStatus.RUNNING:
-            return State.RUNNING
-        else:
-            self.log.error('Event: Invalid state %s on job %s', status, job_id)
-            return State.FAILED
+"""This module is deprecated. Please use `kubernetes.client.models for V1ResourceRequirements and Port."""
+# flake8: noqa
+# pylint: disable=unused-import
+import warnings
+
+from airflow.kubernetes import pod_launcher_deprecated  # pylint: disable=unused-import
```

Comment:
This does not take care of the case if someone would have been importing class from `airflow/kubernetes/pod_launcher.py`

example:

```python
>>> from airflow.kubernetes.pod_launcher import PodStatus, PodLauncher
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'PodStatus'
>>> from airflow.kubernetes.pod_launcher import PodLauncher
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'PodLauncher'
```

This will now raise AttributeNotFoundError

Reply:
Please also rebase on latest master

Reply:
@kaxil I changed the import so this should be fixed.

Reply:
Looks good now.

Reply:
Cool